matcher/service/openreview_interface.py

```diff
@@ -27,16 +25,42 @@ def __init__(self, client, config_note_id, logger=logging.getLogger(__name__)):
         self._demands = None
         self._constraints = None
 
-        self.validate_score_spec()
-
-    def validate_score_spec(self):
+        self._init_config_attributes(config_note_id)
+        self._validate_score_spec()
+
+    def _init_config_attributes(self, config_note_id):
+        try:
+            self.logger.debug('GET note id={}'.format(config_note_id))
+            self.config_note = self.client.get_note(config_note_id)
+        except openreview.OpenReviewException as error_handle:
+            self.logger.error('Config note note found: {}'.format(config_note_id))
+            raise ConfigNoteInterfaceError('Config note not found') from error_handle
+
+        try:
+            self.logger.debug('GET invitation id={}'.format(self.config_note.content['assignment_invitation']))
+            self.assignment_invitation = self.client.get_invitation(self.config_note.content['assignment_invitation'])
+        except openreview.OpenReviewException as error_handle:
+            self.logger.error('Assignment invitation not found: {}'.format(self.config_note.content['assignment_invitation']))
+            raise ConfigNoteInterfaceError('Assignment invitation not found') from error_handle
+
+        try:
+            self.logger.debug('GET invitation id={}'.format(self.config_note.content['aggregate_score_invitation']))
+            self.aggregate_score_invitation = self.client.get_invitation(self.config_note.content['aggregate_score_invitation'])
+        except openreview.OpenReviewException as error_handle:
+            self.logger.error('Aggregate score invitation not found: {}'.format(self.config_note.content['aggregate_score_invitation']))
+            raise ConfigNoteInterfaceError('Aggregate score invitation not found') from error_handle
+
+        self.num_alternates = int(self.config_note.content.get('alternates', 0))
+
+    def _validate_score_spec(self):
         for invitation_id in self.config_note.content.get('scores_specification', {}):
+            self.logger.debug('GET invitation id={}'.format(invitation_id))
             try:
-                self.logger.debug('GET invitation id={}'.format(invitation_id))
                 self.client.get_invitation(invitation_id)
             except openreview.OpenReviewException as error_handle:
+                self.logger.error('Score invitation not found: {}'.format(invitation_id))
                 self.set_status(MatcherStatus.ERROR)
```

Comment:
this set_status shouldn't be here. The datasource shouldn't decide the result of the matching, if there is an exception getting the data then the caller is responsable of deciding what to do next.

Reply:
Fixed.